src/library/Cardano/CoinSelection.hs

```diff
@@ -240,7 +210,7 @@ data CoinSelectionError e
     -- available UTxO entries were depleted before all the requested
     -- transaction outputs could be paid for.
     --
-    | ErrMaximumInputCountExceeded Word64
+    | ErrMaximumInputCountExceeded Integer
```

Comment:
the same remark/question like above?

Reply:
Well spotted. I've changed this (and other similar values) to use `Natural`.

Fixed in cc259b15b77499ce80a101608b44f55471b830b0.